testing/endpoint.go

```diff
@@ -0,0 +1,394 @@
+package ibctesting
+
+import (
+	"fmt"
+
+	//	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	connectiontypes "github.com/cosmos/ibc-go/modules/core/03-connection/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	commitmenttypes "github.com/cosmos/ibc-go/modules/core/23-commitment/types"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	"github.com/cosmos/ibc-go/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
+	"github.com/cosmos/ibc-go/testing/mock"
+)
+
+// Endpoint
+type Endpoint struct {
+	Chain        *TestChain
+	Counterparty *Endpoint
+	ClientID     string
+	ConnectionID string
+	ChannelID    string
+	PortID       string
+
+	ClientConfig     ClientConfig
+	ConnectionConfig *ConnectionConfig
+	ChannelConfig    *ChannelConfig
+}
+
+func NewEndpoint(chain *TestChain) *Endpoint {
```

Comment:
```suggestion
func NewDefaultEndpoint(chain *TestChain) *Endpoint {
```

Then create a constructor `NewEndpoint` that allows you to pass in each field